x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/conditional/CaseExtraTests.java

```diff
@@ -72,6 +72,19 @@ public void testPartialFoldDropsFirstFalse() {
         );
     }
 
+    public void testPartialFoldMv() {
+        Case c = new Case(
+            Source.synthetic("case"),
+            new Literal(Source.EMPTY, List.of(true, true), DataType.BOOLEAN),
```

Comment:
- Maybe let's randomize the booleans and the number of MVs?
- Maybe let's also consider cases with multiple multivalued conditions, not just one?
- Also maybe let's add cases where `partiallyFold` can fully fold, and returns the same as `fold`?

Reply:
The first two I'm doing now. The last one I'll have a think about. I think I'd like to make a method to test partiallFold inside `CaseTests` so it can use the test cases.

Reply:
I've added a bunch of cases.